src/omlt/neuralnet/nn_formulation.py

```diff
@@ -166,8 +177,70 @@ def output_assignment(b, *output_index):
         return b.scaled_outputs[output_index] == b.layer[id(output_layer)].z[output_index]
 
 
-
-class ReducedSpaceNeuralNetworkFormulation(_PyomoFormulation):
+class FullSpaceSmoothNNFormulation(FullSpaceNNFormulation):
+    def __init__(self, network_structure):
+        """
+        This class is used for building "full-space" formulations of 
+        neural network models composed of smooth activations (e.g., tanh,
+        sigmoid, etc.)
+
+        Parameters
+        ----------
+        network_structure : NetworkDefinition
+           the neural network definition
+        """
+        super().__init__(network_structure)
+
+    def _supported_default_activation_constraints(self):
+        return {
+            "linear": linear_activation_constraint,
+            "sigmoid": sigmoid_activation_constraint,
+            "softplus": softplus_activation_constraint,
+            "tanh": tanh_activation_constraint
+        }
```

Comment:
Possible discussion item: query from `_DEFAULT_ACTIVATION_CONSTRAINTS` where we can. Might be easier to maintain when we add new default functions. we could just pop the 'relu'.